src/qibo/derivative.py

```diff
@@ -115,3 +115,85 @@ def circuit(nqubits = 1):
     circuit.set_parameters(original)
 
     return generator_eigenval * (forward - backward)
+
+
+def rescaled_parameter_shift(
+    circuit,
+    hamiltonian,
+    parameter_index,
+    scale_factor,
+    initial_state=None,
+):
+    """In this method a rescaled version of the parameter shift rule (PSR) is implemented.
+    It is useful when a circuit's parameter is obtained by combination of a variational
+    parameter and an external object, such as a training variable in a Quantum
+    Machine Learning problem. For example, performing a re-uploading strategy
+    to embed some data into a circuit, we apply to the quantum state rotations
+    whose angles are in the form: theta' = theta * x, where theta is a variational
+    parameter and x an input variable. The PSR allows to calculate the derivative
+    with respect of theta' but, if we want to optimize a system with respect its
+    variational parameters we need to "free" this procedure from the x depencency.
+    This is the aim of this method.
+
+    Original references:
+        `https://arxiv.org/abs/1811.11184`;
+        `https://arxiv.org/abs/2210.10787`.
+
+    Args:
+        circuit (:class:`qibo.models.circuit.Circuit`): custom quantum circuit.
+        hamiltonian (:class: `qibo.hamiltonians.Hamiltonian`): target observable.
+        parameter_index (int): the index which identifies the target parameter in the circuit.get_parameters() list
+        scale_factor (float): parameter scale factor;
+        initial_state ((1, 2**nqubits) matrix): initial state on which the circuit acts.
```

Comment:
```suggestion
        initial_state ((2**nqubits,) matrix): initial state on which the circuit acts.
```
as before.

Reply:
With the new code, and following @DiegoGM91 suggestion, this parameter is now represented as `(2**qubits) vector`.